// @nuxtjs/leaflet/dist/module.mjs
import { defineNuxtModule, addComponent } from '@nuxt/kit';

const components = [
  "LCircle",
  "LCircleMarker",
  "LControl",
  "LControlAttribution",
  "LControlLayers",
  "LControlScale",
  "LControlZoom",
  "LFeatureGroup",
  "LGeoJson",
  "LIcon",
  "LImageOverlay",
  "LLayerGroup",
  "LMap",
  "LMarker",
  "LPolygon",
  "LPolyline",
  "LPopup",
  "LRectangle",
  "LTileLayer",
  "LTooltip",
  "LWmsTileLayer"
];
const module = defineNuxtModule({
  meta: {
    name: "nuxt-leaflet",
    configKey: "nuxtLeaflet",
    compatibility: {
      nuxt: ">=3.0.0"
    }
  },
  // Default configuration options of the Nuxt module
  defaults: {},
  async setup(options, nuxt) {
    nuxt.options.css.push("leaflet/dist/leaflet.css");
    for (const component of components) {
      addComponent({
        name: component,
        export: component,
        filePath: "@vue-leaflet/vue-leaflet",
        chunkName: `nuxt-leaflet/${component}`,
        mode: "all"
      });
    }
  }
});

export { components, module as default };
